### DISCUSSION TITLE
[Native DLL] How to get `argc`, `argv` of PascalABC.NET program?

### DISCUSSION THREAD

**Discussion by ghost** [pascalabcnet/pascalabcnet#3009] [General]
The native library defines it own even loop, that needs to be initialized with a function that take `argc` and `argv` as arguments. I don't know how to get `argc`, `argv` of PascalABC.NET program. Please help me. Thank you.

**Comment by SunSerega** [comment]
Do you need to get them, or make arguments that would accept those values passed to you from library code?

In the second case, you can just make an `IntPtr` argument instead of a pointer, and never touch it, since I doubt you'll actually need those values.

**Reply by ghost** [reply]
I think I need to get `argc` and `argv` then pass them to the external function. Imagine something like this: `myprog.exe --input test.txt --overwrite`. The `argc` and `argv` need to be passed to the external function in order for it to work.

**Reply by SunSerega** [reply]
I'm pretty sure https://github.com/pascalabcnet/pascalabcnet/discussions/3011 is a dupe of this discussion. Well, I've answered there already.
